zflecs: set_pair compiler error

```
$ zig version
0.12.0-dev.140+e078324db
```
I tried adding a [test for zflecs `ecs.set_pair`](https://github.com/Pyrolistical/zig-gamedev/blob/e7961710eab7ff0ce94f099053b5698088c3c53a/libs/zflecs/src/tests.zig#L313-L329):
```zig
const world = ecs.init();
defer _ = ecs.fini(world);

const Smiles = struct {};
ecs.TAG(world, Walking);
ecs.TAG(world, Smiles);

const awkwardly = ecs.new_entity(world, "Awkwardly");

const entity = ecs.new_entity(world, "Bob");
_ = ecs.set_pair(world, entity, ecs.id(Walking), awkwardly, Walking, .{});
...
```

but that fails to compile:
```
$ zig build test
zig test zflecs-tests Debug native: error: the following command failed with 1 compilation errors:
.../zig test .../zig-gamedev/libs/zflecs/src/zflecs.zig .../zig-gamedev/libs/zflecs/zig-cache/o/bd502a748b6380831be400ece6dd3486/libzflecs.a -lc --cache-dir .../zig-gamedev/libs/zflecs/zig-cache --global-cache-dir .../.cache/zig --name zflecs-tests --mod zflecs::.../zig-gamedev/libs/zflecs/src/zflecs.zig --deps zflecs -I .../zig-gamedev/libs/zflecs/libs/flecs --listen=- 
Build Summary: 1/4 steps succeeded; 1 failed (disable with --summary none)
test transitive failure
└─ run zflecs-tests transitive failure
   └─ zig test zflecs-tests Debug native 1 errors
src/zflecs.zig:2201:94: error: cast increases pointer alignment
    return ecs_set_id(world, subject, pair(first, second), @sizeOf(T), @as(*const anyopaque, @ptrCast(&val)));
                                                                                             ^~~~~~~~~~~~~~
src/zflecs.zig:2201:103: note: '*const tests.Walking' has alignment '0'
    return ecs_set_id(world, subject, pair(first, second), @sizeOf(T), @as(*const anyopaque, @ptrCast(&val)));
                                                                                                      ^~~~
src/zflecs.zig:2201:94: note: '*const anyopaque' has alignment '1'
src/zflecs.zig:2201:94: note: use @alignCast to assert pointer alignment
referenced by:
    test.zflecs.pairs: src/tests.zig:324:21
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

which is erroring within [`ecs.set_pair`](https://github.com/michal-z/zig-gamedev/blob/08550b1d26898a90276254d90801e117a5be10d6/libs/zflecs/src/zflecs.zig#L2193-L2202)

--- a/libs/zflecs/src/tests.zig
+++ b/libs/zflecs/src/tests.zig
@@ -308,3 +308,20 @@ test "zflecs.try_different_alignments" {
         _ = ecs.get(world, entity, Component);
     }
 }
+
+test "zflecs.pairs" {
+    const world = ecs.init();
+    defer _ = ecs.fini(world);
+
+    const Speed = u8;
+    ecs.COMPONENT(world, Speed);
+    ecs.TAG(world, Walking);
+
+    const entity = ecs.new_entity(world, "Bob");
+
+    _ = ecs.set_pair(world, entity, ecs.id(Speed), ecs.id(Walking), Speed, 2);
+    try expect(ecs.has_pair(world, entity, ecs.id(Speed), ecs.id(Walking)));
+
+    _ = ecs.remove_pair(world, entity, ecs.id(Speed), ecs.id(Walking));
+    try expect(!ecs.has_pair(world, entity, ecs.id(Speed), ecs.id(Walking)));
+}

--- a/libs/zflecs/src/zflecs.zig
+++ b/libs/zflecs/src/zflecs.zig
@@ -2198,7 +2198,16 @@ pub fn set_pair(
     comptime T: type,
     val: T,
 ) entity_t {
-    return ecs_set_id(world, subject, pair(first, second), @sizeOf(T), @as(*const anyopaque, @ptrCast(&val)));
+    return ecs_set_id(world, subject, pair(first, second), @sizeOf(T), @as(*const anyopaque, @ptrCast(@alignCast(&val))));
+}
+
+pub fn has_pair(
+    world: *world_t,
+    subject: entity_t,
+    first: entity_t,
+    second: entity_t,
+) bool {
+    return ecs_has_id(world, subject, pair(first, second));
 }
 
 pub fn remove_pair(world: *world_t, subject: entity_t, first: entity_t, second: entity_t) void {

